sklearn/covariance/_shrunk_covariance.py

```diff
@@ -80,23 +80,23 @@ def _oas(X, *, assume_centered=False):
     }
 )
 def shrunk_covariance(emp_cov, shrinkage=0.1):
-    """Calculate a covariance matrix shrunk on the diagonal.
+    """Calculate covariance matrices shrunk on the diagonal.
 
     Read more in the :ref:`User Guide <shrunk_covariance>`.
 
     Parameters
     ----------
-    emp_cov : array-like of shape (n_features, n_features)
-        Covariance matrix to be shrunk.
+    emp_cov : array-like of shape (..., n_features, n_features)
```

Comment:
```suggestion
    emp_cov : array-like of shape (n_features, n_features) or \
            (n_matrices, n_features, n_features)
```

Reply:
Suggestions about docstring seem to restrict usage to 2D and 3D arrays, whereas function is now able to process any nd array.

Reply:
What is the use case of n-D shrinkage? Since the covariance matrix is always a 2-D matrix, I don't really see when you will pass a 4-D array, for instance.

Reply:
Most of NumPy and SciPy functions use `(..., n, n)` to indicate that they can process ndarrays, 
like [numpy.linalg.eig](https://numpy.org/doc/stable/reference/generated/numpy.linalg.eig.html) and [scipy.linalg.expm](https://docs.scipy.org/doc/scipy/reference/generated/scipy.linalg.expm.html) for example.

Use cases belong to users. But, for an example with a 4D array, shape `(k, m, n, n)`, one might want to shrunk `k` sets, each set containing of `m` covariance matrices.

For me, description should not restrain actual usage. But, as you wish.

Reply:
I was thinking more of restraining on purpose the usage that we have in scikit-learn.
But we can go this road and see what other reviewers think.